apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryInsertRow.cs

```diff
@@ -185,13 +196,35 @@ private static object ConvertRowValue(object value)
             if (value is DateTime)
             {
                 DateTime dt = (DateTime)value;
-                // TODO: Maybe enforce universal only?
-                return dt.ToUniversalTime().ToString("yyyy-MM-dd HH:mm:ss.FFFFFF'Z'", CultureInfo.InvariantCulture);
+                switch (dt.Kind)
+                {
+                    // Civil datetime
+                    case DateTimeKind.Unspecified:
+                        return dt.ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFF", CultureInfo.InvariantCulture);
+                    // Timestamp
+                    case DateTimeKind.Utc:
+                        return dt.ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFF'Z'", CultureInfo.InvariantCulture);
+                    // We can get into this situation if we accept a list of DateTime values. We can't
+                    // validate when we're given the list, as it can change.
+                    default:
+                        throw new InvalidOperationException("Local DateTime values are not supported");
+                }
             }
             else if (value is DateTimeOffset)
             {
                 DateTimeOffset dto = (DateTimeOffset)value;
-                return dto.ToUniversalTime().ToString("yyyy-MM-dd HH:mm:ss.FFFFFF'Z'", CultureInfo.InvariantCulture);
+                // Timestamp
+                return dto.ToUniversalTime().ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFF'Z'", CultureInfo.InvariantCulture);
+            }
+            else if (value is TimeSpan)
+            {
+                TimeSpan ts = (TimeSpan) value;
+                if (ts < TimeSpan.Zero || ts >= TimeSpan.FromHours(24))
```

Comment:
Yup - it's Time, as in time-of-day.
https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#time-type